MeshGeoToolsLib/GeoMapper.cpp

```diff
@@ -212,208 +217,343 @@ double GeoMapper::getMeshElevation(
     return (*(_surface_mesh->getNode(pnt->getID())))[2];
 }
 
-std::unique_ptr<std::vector<GeoLib::Polyline*>> copyPolylinesVector(
-    std::vector<GeoLib::Polyline*> const& polylines,
-    std::vector<GeoLib::Point*> const& points)
+/// Find the 2d-element within the \c elements that contains the given point \c p.
+///
+/// The algorithm projects every element of the elements vector and the point
+/// \c p orthogonal to the \f$x\f$-\f$y\f$ plane. In the \f$x\f$-\f$y\f$ plane
+/// it is checked if the projeted point is in the projected element.
+static MeshLib::Element const* findElementContainingPoint2d(
```

Comment:
Maybe better call this one ...XY instead of 2d, because the latter is not unique...